x-pack/plugin/esql/qa/testFixtures/src/main/resources/spatial.csv-spec

```diff
@@ -49,6 +87,136 @@ WIIT           | POINT(105.176060419161 -5.242566777132)   | Radin Inten II
 ZAH            | POINT(60.900708564915 29.4752941956573)   | Zahedan Int'l               | 9           | mid
 ;
 
+centroidFromAirports#[skip:-8.12.99, reason:st_centroid added in 8.13]
+// tag::st_centroid-airports[]
+FROM airports
+| STATS centroid=ST_CENTROID(location)
+// end::st_centroid-airports[]
+;
+
+// tag::st_centroid-airports-result[]
+centroid:geo_point
+POINT(-0.030548143003023033 24.37553649504829)
+// end::st_centroid-airports-result[]
+;
+
+centroidFromAirportsNested#[skip:-8.12.99, reason:st_centroid added in 8.13]
+FROM airports
+| STATS centroid=ST_CENTROID(TO_GEOPOINT(location))
+;
+
+centroid:geo_point
+POINT (-0.03054810272375508 24.37553651570554)
+;
+
+centroidFromAirportsCount#[skip:-8.12.99, reason:st_centroid added in 8.13]
+FROM airports
+| STATS centroid=ST_CENTROID(location), count=COUNT()
+;
+
+centroid:geo_point                             | count:long
+POINT(-0.030548143003023033 24.37553649504829) | 891
+;
+
+centroidFromAirportsCountGrouped#[skip:-8.12.99, reason:st_centroid added in 8.13]
+FROM airports
+| STATS centroid=ST_CENTROID(location), count=COUNT() BY scalerank
+| SORT scalerank DESC
+;
+
+centroid:geo_point                            | count:long | scalerank:i
```

Comment:
Not related to this specific test, but for a more wide range of value handlers some `null` values could be added to the "airport" data set.

Reply:
Added more data with null values and have aggregation working on that. The centroid of null is null. The centroid of data including a few nulls is the centroid of the non-null values.